spec/models/miq_report/generator_spec.rb

```diff
@@ -67,6 +67,23 @@
         expect(@miq_report_profile_all.table.data[0].data).to include("min_trend_value" => 400,
                                                                       "max_trend_value" => 700)
       end
+
+      it "handles a basic report with a where_clause and a where clause passed in options" do
+        FactoryBot.create(:vm)       # filtered out by option[:where_clause]
+        FactoryBot.create(:template) # filtered out by report.where_clause
+        vm = FactoryBot.create(:vm, :vendor => "redhat")
+
+        rpt = FactoryBot.create(
+          :miq_report,
+          :db           => "VmOrTemplate",
+          :where_clause => ["vms.type = ?", "Vm"],
+          :col_order    => %w[id name host.name vendor]
+        )
+        rpt.generate_table(:userid => @user.userid, :where_clause => {:vendor => "redhat"})
```

Comment:
This does test different formats (string and hash)

I didn't see the need to pass in more varying options since rails already has those options in their test suite